### PR TITLE
ipc4: add fw ready support for ipc4

### PR DESCRIPTION
**Pull request by RanderWang**
Add ipc_boot_complete_msg in ipc3 & ipc4

### CODE REVIEW

File: src/ipc/handler-ipc4.c [hunk 1]
```diff
@@ -239,6 +239,12 @@ ipc_cmd_hdr *ipc_compact_read_msg(void)
 	return ipc_to_hdr(msg_in);
 }
 
+int ipc_boot_complete_msg(uint32_t *header, , uint32_t *data)
```
Comment (lgirdwood) [inline src/ipc/handler-ipc4.c:242]: two commas here.

File: src/include/sof/ipc/common.h [hunk 2]
```diff
@@ -139,6 +139,14 @@ int ipc_process_host_buffer(struct ipc *ipc,
  */
 int ipc_dma_trace_send_position(void);
 
+/**
+ * \brief create a IPC boot completet message.
+ * @header[in] msg header.
+ * @data[in] msg data.
+ * @return Number of words written.
+ */
+int ipc_boot_complete_msg(uint32_t *header, uint32_t *data);
```
Comment (lgirdwood) [inline src/include/sof/ipc/common.h:148]: We should really use the new ```ipc_cmd_hdr``` type here instead of uint32_t.

**Review by lgirdwood — CHANGES_REQUESTED** [review]
This may have to block on the ipc abstraction being merged. Can you do on top of the ipc4 branch.

**Comment by jgunn0262** [comment]
@RanderWang I'm assuming there is nothing forcing me to use IPC4 in here ?

**Comment by RanderWang** [comment]
> @RanderWang I'm assuming there is nothing forcing me to use IPC4 in here ?
we can select it in kconfig

**Comment by RanderWang** [comment]
wrong branch

**Pull request closed without merging**